### ISSUE TITLE
Support for wrappers like sudo

### ISSUE THREAD

**Issue by maralorn** [luke-clifton/shh#24]
I love to use shh to have compile-time checks for the existence of called binaries.
When I use `sudo "command"` I loose a lot of that. It feels like an inherent problem.
Do you have any ideas for a workaround existing or easily implementable anyways?

**Comment by luke-clifton** [comment]
Indeed! The `bigrejig` branch has some support for this.

I've been holding off on releasing it until I have a bit more experience with it. I haven't used it for `sudo`, but I have used it for `env`.

**Comment by maralorn** [comment]
I just realized yesterday that at least under nixos things get more complicated because (quite obviously) `sudo` can only be used if installed in the system-path. So introducing it into scope via `pkgs.sudo` won’t help. So right now I am at `exe "sudo" "command"`. But that works so, no worries.

**Comment by luke-clifton** [comment]
So, with the `bigrejig` branch, you could do

```haskell
{-# LANGUAGE RankNTypes, TemplateHaskell #-}
module Main where
import Shh

loadEnv Absolute

mysudo :: Cmd -> Cmd
mysudo cmd = exe "sudo" (displayCommand cmd)
```

And use it like...
```
λ mysudo whoami
root
λ whoami
lukec
```

It unfortunately requires that ugly `displayCommand` bit, so making a wrapper like that proves to be the most user friendly. I'm still trying to work it so that I don't need that without completely ruining type inference.

I'll ping you here once I decide to release that version.

**Comment by luke-clifton** [comment]
@maralorn just released 0.7.0.0 on hackage. It's got a few breaking changes, but supports the pattern I provided above.
